src/fu-engine.c

```diff
@@ -1079,11 +1079,29 @@ static gboolean
 fu_engine_check_requirement_firmware (FuEngine *self, XbNode *req,
 				      FuDevice *device, GError **error)
 {
+	FuDevice *device_actual = device;
+	guint64 depth;
 	g_autoptr(GError) error_local = NULL;
 
+	/* look at the parent device */
+	depth = xb_node_get_attr_as_uint (req, "depth");
+	if (depth != G_MAXUINT64) {
+		for (guint i = 0; i < depth; i++) {
```

Comment:
Doesn't this need to be guint64 to make sure you compare same types?

Reply:
Good catch, although I think it's unlikely to have more than 4294967295 nested children :)

Reply:
I sure hope not 😆